diff --git a/src/simulator/util/dummy_perception_publisher/src/node.cpp b/src/simulator/util/dummy_perception_publisher/src/node.cpp
index 3f92dd6e..43fbd1ae 100644
--- a/src/simulator/util/dummy_perception_publisher/src/node.cpp
+++ b/src/simulator/util/dummy_perception_publisher/src/node.cpp
@@ -271,6 +271,8 @@ void DummyPerceptionPublisherNode::pedestrianPoseCallback(const geometry_msgs::P
 {
     std::tuple<geometry_msgs::PoseStamped, geometry_msgs::TwistStamped, int> tuple;
     convert2Tuple(input_pose_msg, input_twist_msg, object_id, tuple);
+    std::get<0>(tuple).header.stamp = ros::Time::now();
+    std::get<1>(tuple).header.stamp = ros::Time::now();
     pedestrian_poses_.push_back(tuple);
 }
 
@@ -279,6 +281,8 @@ void DummyPerceptionPublisherNode::carPoseCallback(const geometry_msgs::PoseStam
 {
     std::tuple<geometry_msgs::PoseStamped, geometry_msgs::TwistStamped, int> tuple;
     convert2Tuple(input_pose_msg, input_twist_msg, object_id, tuple);
+    std::get<0>(tuple).header.stamp = ros::Time::now();
+    std::get<1>(tuple).header.stamp = ros::Time::now();
     car_poses_.push_back(tuple);
 }
 
